lua/haskell-tools/lsp/eval.lua

```diff
@@ -0,0 +1,58 @@
+---@mod haskell-tools.lsp.eval LSP code snippet evaluation
+local eval = {}
+local lsp_util = require('haskell-tools.lsp.util')
+
+---@param list any[]
+---@return any[] reversed
+local function reverse_list(list)
+  local rev = {}
+  for i = #list, 1, -1 do
+    rev[#rev + 1] = list[i]
+  end
+  return rev
+end
+
+---@param bufnr number
+local function get_eval_command_lenses(bufnr, exclude_lines)
+  exclude_lines = exclude_lines or {}
+  local eval_cmd_lenses = {}
+  for _, lens in pairs(vim.lsp.codelens.get(bufnr)) do
+    if lens.command.command:match('evalCommand') and not vim.tbl_contains(exclude_lines, lens.range.start.line) then
+      eval_cmd_lenses[#eval_cmd_lenses + 1] = lens
```

Comment:
TODO:
Instead of reversing the list, use `table.insert(eval_cmd_lenses, 1, lens)`